tests/test_jsonschema2md.py

```diff
@@ -580,3 +580,62 @@ def test_pattern_ignore(self):
             'Default: `"infer"`.\n',
         ]
         assert expected_output == parser.parse_schema(test_schema)
+
+    def test_collapse_children(self):
+        test_schema = {
+            "type": "object",
+            "properties": {
+                "general": {
+                    "description": "General settings.",
+                    "type": "object",
+                    "additionalProperties": False,
+                    "properties": {
+                        "pipeline": {
+                            "description": "Pipeline to use, depending on input format",
+                            "type": "object",
+                            "additionalProperties": False,
+                            "properties": {
+                                "foo": {
+                                    "description": "Foo description",
+                                    "type": "string",
+                                    "enum": [
+                                        "infer",
+                                        "pin",
+                                        "tandem",
+                                        "maxquant",
+                                        "msgfplus",
+                                        "peptideshaker",
+                                    ],
+                                    "default": "infer",
+                                }
+                            },
+                        },
+                    },
+                }
+            },
+        }
+        parser = jsonschema2md.Parser(collapse_children=True)
+        expected_output = [
+            "# JSON Schema\n\n",
+            "## Properties\n\n",
+            "<details>\n",
+            "<summary>",
+            '<a id="properties/general"></a><strong><code>general</code></strong> '
+            "<em>(object)</em>: General settings. Cannot contain additional "
+            "properties.",
+            "</summary>\n\n",
+            "<details>\n",
+            "<summary>",
+            "<a "
+            'id="properties/general/properties/pipeline"></a><strong><code>pipeline</code></strong> '
+            "<em>(object)</em>: Pipeline to use, depending on input format. Cannot "
+            "contain additional properties.",
+            "</summary>\n\n",
```

Comment:
This double newline is needed so that the text below is interpreted as markdown.